app/javascript/shared/components/ui/MutiselectDropdown.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <div v-on-clickaway="onCloseDropdown" class="dropdown-wrap">
+    <button
+      :v-model="selectedItem"
+      class="button-input"
+      @click="toggleDropdown"
+    >
+      <Thumbnail
+        v-if="isValueExist"
+        :src="selectedItem.thumbnail"
+        size="24px"
+        :status="selectedItem.availability_status"
+        :badge="selectedItem.channel"
+        :username="selectedItem.name"
+      />
+      <div class="name-icon-wrap">
+        <div v-if="!isValueExist" class="name select">
+          {{ multiselectorPlaceholder }}
+        </div>
+        <div v-else class="name" :title="selectedItem.name">
+          {{ selectedItem.name }}
+        </div>
+        <i v-if="showSearchDropdown" class="icon ion-chevron-up" />
+        <i v-else class="icon ion-chevron-down" />
+      </div>
+    </button>
+    <div
+      :class="{ 'dropdown-pane--open': showSearchDropdown }"
+      class="dropdown-pane"
+    >
+      <h4 class="text-block-title">
+        {{ multiselectorTitle }}
+      </h4>
+      <multiselect-dropdown-items
+        v-if="showSearchDropdown"
+        :options="options"
+        :selected-item="selectedItem"
+        :input-placeholder="inputPlaceholder"
+        :no-search-result="noSearchResult"
+        @click="onClickSelectItem"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import Thumbnail from 'dashboard/components/widgets/Thumbnail.vue';
+import MultiselectDropdownItems from 'shared/components/ui/MultiselectDropdownItems';
+import { mixin as clickaway } from 'vue-clickaway';
+export default {
+  components: {
+    Thumbnail,
+    MultiselectDropdownItems,
+  },
+  mixins: [clickaway],
+  props: {
+    options: {
+      type: Array,
+      default: () => [],
+    },
+    selectedItem: {
+      type: Object,
+      default: () => ({}),
+    },
+    multiselectorTitle: {
+      type: String,
+      default: 'Select',
+    },
+    multiselectorPlaceholder: {
+      type: String,
+      default: 'None',
+    },
+    noSearchResult: {
+      type: String,
+      default: 'No results found',
+    },
+    inputPlaceholder: {
+      type: String,
+      default: 'Search',
+    },
+  },
+  data() {
+    return {
+      showSearchDropdown: false,
+    };
+  },
+  computed: {
+    isValueExist() {
+      if (
+        this.selectedItem &&
+        this.selectedItem.name &&
+        this.selectedItem &&
+        this.selectedItem.id
```

Comment:
Used `this.selectedItem` multiple times. Can we use a condition something like`this.selectedItem && this.selectedItem.id`